backend/controllers/artworks.ts

```diff
@@ -1,91 +1,115 @@
-    } catch (error) {
-        next(error)
+    } else {
+        const err = new Error(`Incorrect request body provided`)
+        res.status(400)
+        return next(err)
     }
-})
+    
+}))
 
-const editArtwork = (async (req: Request, res: Response, next: NextFunction) => {
-    try {
-        const token = req.headers.authorization?.split(" ")[1]
-        if (!token) return res.status(401).json({ error: 'Access denied' });
+const editArtwork = authAsyncWrapper((async (req: Request, res: Response, next: NextFunction) => {
+    const artworkId = req.params.artworkId
+    if(req.body.categories !== undefined && req.body.collectionName !== undefined) {
         try {
-            const decoded = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET as string)
-            const editedArtwork = await Artwork.replaceOne({_id: req.params.artworkId}, req.body)
-            return res.status(201).json(editedArtwork)
-        } catch (error) {
-            return res.status(401).json({ error: 'Access denied' });
+            const editedArtwork = await Artwork.replaceOne({_id: artworkId}, req.body).exec()
+            if(editedArtwork.modifiedCount === 0) {
+                const err = new Error(`Artwork not found`)
+                res.status(404)
+                return next(err)
+            } else {
+                return res.status(201).json(editedArtwork)
+            }
+        } catch {
+            const err = new Error(`Database unavailable`)
+            res.status(503)
+            return next(err)
         }
-    } catch (error) {
-        next(error)
+    } else {
+        const err = new Error(`Incorrect request body provided`)
+        res.status(400)
+        return next(err)
     }
-})
-
-const deleteArtworks = async (req: Request, res: Response, next: NextFunction) => {
-    try {
-        const token = req.headers.authorization?.split(" ")[1]
-        if (!token) return res.status(401).json({ error: 'Access denied' });
-        try {
-            const decoded = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET as string)
-            const artworksToDelete = req.params.artwork
-            if (!artworksToDelete) {
-                return res.status(400).send({ message: "Artworks not found" })
-            }
-            const artworksToDeleteList = artworksToDelete.split(",")
-            const existingArtworks = await Artwork.find({ _id: { $in: artworksToDeleteList } })
-
-            if (existingArtworks.length === 0) {
-                return res.status(404).send({ message: "Artworks not found" })
-            }
+}))
 
-            const result = await Artwork.deleteMany({ _id: { $in: artworksToDeleteList } })
-
-            res.status(200).json({ message: req.params.artwork, deletedCount: result.deletedCount })
-        } catch (error) {
-            return res.status(401).json({ error: 'Access denied' });
-        }
+const deleteArtworks = authAsyncWrapper(async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Tutaj AI fajnie zaproponował refactor tego, żeby nie robić tyle try catchy jeden w drugim, bo to trochę bad pattern jest: 
```js 
const deleteArtworks = authAsyncWrapper(async (req: Request, res: Response, next: NextFunction) => {
  const { ids } = req.body;
  if (!ids) {
    return res.status(400).json({ error: "Incorrect request body provided" });
  }

  const session = await mongoose.startSession();
  try {
    await session.withTransaction(async () => {
      const artworksToDelete = ids;
      if (!Array.isArray(artworksToDelete) || artworksToDelete.length === 0) {
        throw new Error("Artworks not specified");
      }

      const count = await Artwork.countDocuments({ _id: { $in: artworksToDelete } }).session(session);
      if (count !== artworksToDelete.length) {
        throw new Error("Artworks not found");
      }

      const result = await Artwork.deleteMany({ _id: { $in: artworksToDelete } }).session(session);
      res.status(200).json(result);
    });
  } catch (error) {
    if (error.message === "Artworks not specified") {
      res.status(400).json({ error: error.message });
    } else if (error.message === "Artworks not found") {
      res.status(404).json({ error: error.message });
    } else {
      res.status(503).json({ error: "Couldn't complete database transaction" });
    }
  } finally {
    session.endSession();
  }
});
```